### PR TITLE
Unparse `SubqueryAlias` without projections to SQL

### PR DESCRIPTION
**Pull request by goldmedal**
## Which issue does this PR close?

Closes #.

## Rationale for this change
Consider the following SQL
```SQL
SELECT 
  c.custkey, 
  c.name 
FROM 
  (
    SELECT 
      c_custkey as custkey, 
      c_name as name 
    FROM 
      customer
  ) c
```
The unoptimized plan is
```
Projection: c.custkey, c.name
  SubqueryAlias: c
    Projection: customer.c_custkey AS custkey, customer.c_name AS name
      TableScan: customer
```
However, after the `optimize_projections` rule, the top-level projection will be removed like
```
SubqueryAlias: c
  Projection: customer.c_custkey AS custkey, customer.c_name AS name
    TableScan: customer projection=[c_custkey, c_name]
```
The unparsing result would like
```SQL
SELECT 
  customer.c_custkey AS custkey, 
  customer.c_name AS "name" 
FROM 
  (
    SELECT 
      customer.c_custkey, 
      customer.c_name 
    FROM 
      customer
  ) AS c
```
It's an invalid SQL because the column qualifier is the table name `customer` but it's in the alias `c` scope actually.

## What changes are included in this PR?
When unparsing a `SubqueryAlias` without any projections, we should create another scope for the alias like
```SQL
SELECT 
  * 
FROM 
  (
    SELECT 
      customer.c_custkey AS custkey, 
      customer.c_name AS "name" 
    FROM 
      (
        SELECT 
          customer.c_custkey, 
          customer.c_name 
        FROM 
          customer
      )
  ) AS c
```

## Are these changes tested?
yes

## Are there any user-facing changes?
The unparsing result is changed.

### CODE REVIEW

File: datafusion/sql/tests/cases/plan_to_sql.rs [hunk 1]
```diff
@@ -697,7 +739,7 @@ fn test_table_scan_pushdown() -> Result<()> {
         plan_to_sql(&table_scan_with_projection_alias)?;
     assert_eq!(
         format!("{}", table_scan_with_projection_alias),
-        "SELECT ta.id, ta.age FROM t1 AS ta"
+        "SELECT * FROM (SELECT ta.id, ta.age FROM t1 AS ta) AS ta"
```
Comment (findepi) [inline datafusion/sql/tests/cases/plan_to_sql.rs:742]: That's not an improvement, right?
I am OK if this is necessary, but is there a different way to fix the original problem, without affecting here?

Comment (goldmedal) [reply datafusion/sql/tests/cases/plan_to_sql.rs]: Thanks @findepi. Sorry for replying so late 🙇 
You're right. I shouldn't break the result of the table scan with any pushdown. I stopped projecting a wildcard if the `SubqueryAlias`'s child is a pushdown table scan.

File: datafusion/sql/tests/cases/plan_to_sql.rs [hunk 2]
```diff
@@ -667,12 +667,95 @@ where
...
+        &schema,
+        None,
+        vec![col("id").gt(lit(1)), col("age").lt(lit(2))],
+    )?
+    .project(vec![col("id")])?
+    .alias("a")?
+    .build()?;
+    let table_scan_with_two_filter = plan_to_sql(&table_scan_with_two_filter)?;
+    assert_eq!(
+        format!("{}", table_scan_with_two_filter),
```
Comment (findepi) [inline datafusion/sql/tests/cases/plan_to_sql.rs:720]: nit: `format!` is unnecessary, just use table_scan_with_two_filter (or table_scan_with_two_filter.to_string()) in assert

(same in other places)

File: datafusion/sql/tests/cases/plan_to_sql.rs [hunk 3]
```diff
@@ -667,12 +667,95 @@ where
...
+
+    let table_scan_with_two_filter = table_scan_with_filters(
+        Some("t1"),
+        &schema,
+        None,
+        vec![col("id").gt(lit(1)), col("age").lt(lit(2))],
+    )?
+    .project(vec![col("id")])?
+    .alias("a")?
+    .build()?;
```
Comment (findepi) [inline datafusion/sql/tests/cases/plan_to_sql.rs:717]: should this be indented?

File: datafusion/sql/tests/cases/plan_to_sql.rs [hunk 4]
```diff
@@ -667,12 +667,95 @@ where
...
+        format!("{}", sql),
+        "SELECT * FROM (SELECT t1.id FROM t1) AS a"
+    );
+
+    let plan = table_scan(Some("t1"), &schema, None)?
+        .filter(col("id").gt(lit(5)))?
+        .project(vec![col("id")])?
+        .alias("a")?
+        .build()?;
+    println!("{}", plan.display_indent());
```
Comment (findepi) [inline datafusion/sql/tests/cases/plan_to_sql.rs:702]: left over?

**Review by findepi — APPROVED** [review]

File: datafusion/sql/tests/cases/plan_to_sql.rs (hunk 3 above)
Comment (goldmedal) [reply datafusion/sql/tests/cases/plan_to_sql.rs:710]: It's formatted by the cargo formatter. I think we can't control it 🤔

File: datafusion/sql/src/unparser/plan.rs [hunk 5]
```diff
@@ -249,12 +248,9 @@ impl Unparser<'_> {
     ) -> Result<()> {
         match plan {
             LogicalPlan::TableScan(scan) => {
-                if scan.projection.is_some()
-                    || !scan.filters.is_empty()
-                    || scan.fetch.is_some()
+                if let Some(unparsed_table_scan) =
```
Comment (alamb) [inline datafusion/sql/src/unparser/plan.rs:251]: Nice

**Review by alamb — APPROVED** [review]
Thank you @goldmedal and @findepi for the review

**Comment by alamb** [comment]
FYI @phillipleblanc and @sgrebnov

**Review by phillipleblanc — APPROVED** [review]
This PR makes sense to me, thanks!

**Comment by sgrebnov** [comment]
LGTM 👍

**Comment by alamb** [comment]
Thanks @sgrebnov @findepi and @phillipleblanc  for the review

**Comment by goldmedal** [comment]
Thanks @alamb @sgrebnov @findepi @phillipleblanc !

**Pull request merged**